flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/LogicalSlot.java

```diff
@@ -152,7 +157,14 @@ public void fail(Throwable cause) {
 		 *
 		 * @param cause of the failure
 		 */
-		void fail(Throwable cause);
+		void failAsync(Throwable cause);
+
+		/**
+		 * Fail the payload with the given cause.
+		 *
+		 * @param cause of the failure
+		 */
+		void failSync(Throwable cause);
```

Comment:
Maybe add a short comment that this is only introduced for the legacy code and should be removed once the legacy code has been removed.

Reply:
You mean I comment on the `failAsync`instead I guess? because that is the one we will always use later once the slot pool also runs in the JM main thread.

Reply:
True.